### DISCUSSION TITLE
Arraylist와 Linkedlist의 차이점을 설명해 주세요.

### DISCUSSION THREAD

**Discussion by BaekYuri** [wanted-9th-3team/frontend-study#54] [최신 기술 및 산업 질문]
각각의 장점과 단점. 속도부분에서도 설명해주세요

**Comment by michoball** [comment]
장점 
 Arraylist -  인덱스를 사용해서 특정 값에 접근할 수 있기 때문에 값을 찾는 속도가 매우 빠릅니다. 반복문 등을 통해 모든 요소를 순차적으로 처리할 때 유리합니다.

linkedlist - 요소의 삽입과 삭제가 간단합니다. 중간에 삽입할 때도 요소들을 이동시킬 필요가 없기 때문에 좋습니다.  동적으로 메모리를 사용합니다. 

단점
Arraylist - 요소의 삽입과 삭제가 배열의 재배치를 필요로 하므로, 중간에 요소를 삽입하거나 삭제하는 경우에는 뒤의 요소들을 이동해야 합니다. 크기를 조정하기위해 추가적인 메모리 할당이 필요합니다. 

linkedlist - 특정 요소에 접근하려면 맨 앞부터 순차적으로 순회를 해야합니다.  데이터를 저장하는 노드이외에 추가적으로 링크를 위한 메모리가 필요합니다.

**Comment by WooGie911** [comment]
Arraylist는 입력된 데이터들이 메모리 공간에서 연속적으로 저장되어 있는 자료구조로 index를 통한 접근이 용이하며 배열의 크기는 처음 생성할 때 정하며 이후에 변경할 수 없습니다.

Linkedlist는 여러 개의 노드들이 순차적으로 연결된 형태를 갖는 자료구조이며, 첫번째 노드를 head, 마지막 노드를 tail이라고 합니다. 각 노드는 데이터와 다음 노드를 가리키는 포인터로 이루어져 있으며 배열과는 다르게 메모리를 연속적으로 사용하지 않습니다. 
Arraylist와 비교 했을 때 무작위 접근, 즉 원하는 위치에 한번에 접근할 수 없어 순차적으로 접근해야 하는 면에서 특정 자료 탐색 부분에서 불리하나, 노드가 연결된 구조이기 때문에 자료의 삽입, 삭제에 용이합니다. 
또한 Tree 구조의 근간이 되는 자료구조이며, Tree 에서 사용되었을 때 그 유용성이 드러납니다.

**Comment by BaekYuri** [comment]
검색 속도 : ArrayList > LinkedList
-> LinkedList는 특정 인덱스를 검색하려면 순차적으로, 단방향으로 접근해야 하기 때문에 많은 시간이 소요될 수 있습니다. ArrayList는 각 index를 가지고 있고, 무작위 접근을 할 수 있기 때문에 검색 속도는 빠른 편입니다.
삽입, 삭제 속도 : ArrayList < LinkedList
-> LinkedList는 데이터를 추가, 삭제할 때 가리키는 노드 값만 수정해주면 되기 때문에 속도가 빠른 편입니다. ArrayList는 추가, 삭제할 때마다 index를 밀어내거나 당겨주는 작업을 해주어야 하기 때문에 느린 편입니다.
